src/algorithm/native/to_wkt.rs

```diff
@@ -0,0 +1,392 @@
+use std::fmt::{Error, Write};
+
+use arrow_array::{builder::GenericStringBuilder, GenericStringArray};
+
+use crate::{
+    array::AsNativeArray, datatypes::GeoDataType, geo_traits::*, trait_::NativeArrayAccessor,
+    NativeArray,
+};
+
+/// Converts the input to WKT representation.
+pub trait ToWKT {
+    type Output;
+
+    fn to_wkt(&self) -> Self::Output;
+}
+
+impl ToWKT for &dyn NativeArray {
+    type Output = Result<GenericStringArray<i32>, Error>;
+
+    fn to_wkt(&self) -> Self::Output {
+        let mut wkt_builder: GenericStringBuilder<i32> = GenericStringBuilder::new();
+
+        match self.data_type() {
+            GeoDataType::Point(_coord_type, _dimension) => todo!(),
+            GeoDataType::LineString(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeLineString(_coord_type, _dimension) => todo!(),
+            GeoDataType::Polygon(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargePolygon(_coord_type, _dimension) => todo!(),
+            GeoDataType::MultiPoint(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeMultiPoint(_coord_type, _dimension) => todo!(),
+            GeoDataType::MultiLineString(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeMultiLineString(_coord_type, _dimension) => todo!(),
+            GeoDataType::MultiPolygon(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeMultiPolygon(_coord_type, _dimension) => todo!(),
+            GeoDataType::Mixed(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeMixed(_coord_type, _dimension) => todo!(),
+            GeoDataType::GeometryCollection(_coord_type, _dimension) => todo!(),
+            GeoDataType::LargeGeometryCollection(_coord_type, _dimension) => todo!(),
+            GeoDataType::WKB => {
+                for item in self.as_wkb().iter() {
+                    match item {
+                        Some(wkb) => {
+                            geometry_to_wkt(&wkb.to_wkb_object(), &mut wkt_builder)?;
+                            wkt_builder.append_value("");
```

Comment:
I'm not exactly sure what how the `GenericStringBuilder` handles the implementation of `Write`. In particular, we need to ensure that it's not adding a new offset for every time `write_str` is called. 

It may be easier to maintain the offsets ourselves in our own builder, but that's easy to change later once the WKT writing has stabilized